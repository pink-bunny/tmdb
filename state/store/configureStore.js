import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

import reducer from './reducer';

const loggerMiddleware = (store) => (next) => (action) => {
  /* eslint-disable no-console */
  console.group(action.type);
  console.info('dispatching', action);
  const result = next(action);
  console.log('next state', store.getState());
  console.groupEnd();
  /* eslint-enable */

  return result;
};

export default function configureStore() {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(reducer, undefined, composeWithDevTools(middlewareEnhancer));

  return store;
}
