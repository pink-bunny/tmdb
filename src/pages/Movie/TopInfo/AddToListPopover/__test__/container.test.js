import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import ConnectedAddToList from '../container';
import {
  fetchMyLists,
  createAndAddToList,
  toggleToList
} from '../../../../../../state/my_lists/actions';

jest.mock('../../../../../../state/my_lists/selectors', () => ({
  myListsList: jest.fn(() => [{ id: 1, name: 'Mock name' }])
}));
jest.mock('../../../../../../state/my_lists/actions', () => ({
  toggleToList: jest.fn(),
  createAndAddToList: jest.fn(),
  fetchMyLists: jest.fn()
}));

describe('AddToList container', () => {
  const store = configureStore()({});
  const props = {
    store,
    movieId: 123
  };
  store.dispatch = jest.fn();
  const wrapper = shallow(<ConnectedAddToList {...props} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */
  const container = wrapper.dive().dive();
  const instance = container.instance();

  it('matches snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('componentDidUpdate()', () => {
    const prevProps = {};
    const prevState = { popoverVisible: false };
    instance.setState({ popoverVisible: true });

    instance.componentDidUpdate(prevProps, prevState);
    expect(fetchMyLists).toHaveBeenCalled();
  });

  it('sets visiblePopover to false', () => {
    instance.visiblePopover(false);
    expect(container.state('popoverVisible')).toBeFalsy();
  });

  it('closes popover', () => {
    const spyVisiblePopover = jest.spyOn(instance, 'visiblePopover');
    instance.closePopover();
    expect(spyVisiblePopover).toHaveBeenCalledWith(false);
  });

  it('creates list and add movie to it', () => {
    const values = { name: 'Mock name', description: 'Mock description' };
    const setErrors = jest.fn();
    const hideModal = jest.fn();
    const { movieId } = props;
    instance.createListModalSubmit(values, setErrors, hideModal);
    expect(createAndAddToList).toHaveBeenCalledWith(values, setErrors, hideModal, movieId);
  });

  it('adds the movie to the list', () => {
    const listId = 123;
    const { movieId } = props;
    instance.addToList(listId);
    expect(toggleToList).toHaveBeenCalledWith(listId, movieId);
  });
});
