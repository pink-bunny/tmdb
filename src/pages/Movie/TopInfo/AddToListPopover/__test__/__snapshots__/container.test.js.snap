// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToList container matches snapshot 1`] = `
<AddToListComponent
  addToList={[Function]}
  closePopover={[Function]}
  handleVisiblePopover={[Function]}
  list={
    Array [
      Object {
        "id": 1,
        "name": "Mock name",
      },
    ]
  }
  popoverCreateListModalSubmit={[Function]}
  popoverVisible={false}
/>
`;
